packages/firestore/test/lite/integration.test.ts

```diff
@@ -2038,3 +2045,142 @@
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a'), limit(1));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(1);
+    });
+  });
+
+  it('test collection count with filter not effected by large limit', () => {
+    //limit that is larger than actual count wouldn't impact the return value
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a'), limit(3));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('test collection count with converter on query', () => {
+    //testing out the converter impact on the AggregateQuery type
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_ = query(collection, where('key', '==', 'a')).withConverter(
+        converter
+      );
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('test collection count with converter on collection', () => {
+    //testing out the converter impact on the AggregateQuery type
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const ref = collection.withConverter(converter);
+      const query_ = query(ref, where('key', '==', 'a'));
+
+      const countQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('aggregateQueryEqual returns true on same queries', () => {
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_1 = query(collection, where('key', '==', 'a'));
+      const query_2 = query(collection, where('key', '==', 'a'));
+
+      const countQuery_1 = countQuery(query_1);
+      const countQuery_2 = countQuery(query_2);
+      expect(aggregateQueryEqual(countQuery_1, countQuery_2)).to.be.true;
+    });
+  });
+  
+  it('aggregateQueryEqual returns false on different queries', () => {
+    return withTestCollectionAndInitialData(testDocs, async collection => {
+      const query_1 = query(collection, where('key', '==', 'a'));
+      const query_2 = query(collection, where('key', '!=', 'a'));
+
+      const countQuery_1 = countQuery(query_1);
+      const countQuery_2 = countQuery(query_2);
+      expect(aggregateQueryEqual(countQuery_1, countQuery_2)).to.be.false;
+    });
+  });
+
+  it('aggregateQuerySnapshotEqual returns true on same queries', () => {
+    return withTestCollectionAndInitialData(testDocs, async collection => {
```

Comment:
Instead of using the naming "original" and "copy", can you use numbers and/or letters? This will help with vertical alignment and improve readability. e.g. name the variables query1A, query1B, query2, etc.